lang/attribute/program/src/declare_program/mods/internal.rs

```diff
@@ -138,17 +140,25 @@ fn gen_internal_accounts_common(
         .iter()
         .flat_map(|ix| ix.accounts.to_owned())
         .collect::<Vec<_>>();
+    let mut account_names: HashSet<String> = std::collections::HashSet::new();
```

Comment:
The [`get_non_instruction_composite_accounts`](https://github.com/coral-xyz/anchor/blob/8dca0d6b2bee5f6e1e3afc797804c485e592f876/lang/attribute/program/src/declare_program/mods/internal.rs#L112-L134) should ideally return a `HashSet` rather than a `Vec`, but the IDL types don't implement the necessary traits currently. I guess I thought [this check](https://github.com/coral-xyz/anchor/blob/8dca0d6b2bee5f6e1e3afc797804c485e592f876/lang/attribute/program/src/declare_program/mods/internal.rs#L126-L128) would make them unique, but it doesn't cover the initial outer case.

I think it would be better for the `get_non_instruction_composite_accounts` to return unique accounts rather than sorting them out here.